src/app/ubs/ubs-admin/components/shared/components/cron-picker/cron-picker.component.ts

```diff
@@ -66,6 +81,57 @@ export class CronPickerComponent implements OnInit, OnDestroy, OnChanges {
     });
   }
 
+  private hourValidator(control: FormControl) {
+    const value = parseInt(control.value, 10);
+    if (isNaN(value) || value < 0 || value >= 25) {
+      return { invalidHour: true };
+    }
+    return null;
+  }
+
+  private timeValidator(control: AbstractControl): null | { [error: string]: boolean } {
+    const hour = control.get('hour')?.value;
+    const min = control.get('min')?.value;
+
+    if (!hour || !min) {
+      return null;
+    }
+    const hourValid = /^[0-2][0-9]$/.test(hour) && hour < 25;
+    const minValid = /^[0-5][0-9]$/.test(min);
+
+    if (hourValid && minValid) {
+      return null;
+    } else {
+      return { invalidHourMin: true };
+    }
+  }
+
+  private isFormControl(control: AbstractControl): control is FormControl {
+    return control instanceof FormControl;
+  }
+
+  getFormControl(controlName: string): FormControl | null {
+    const control = this.form.get(`time.${controlName}`);
+    return this.isFormControl(control) ? control : null;
+  }
+
+  checkForErrorsIn(controlName: string): string | null {
+    const control = this.form.get(`time.${controlName}`);
+
+    if (control?.errors) {
+      if (control.errors.required) {
+        return controlName === 'hour' ? 'cron-picker.errors.hour-required' : 'cron-picker.errors.minute-required';
+      }
+      if (control.errors.pattern) {
```

Comment:
Please refactor this function to reduce its cognitive complexity from 16 to an acceptable 15. 
You might consider combining the following parts to simplify the logic:
```
if (errors.pattern || errors.invalidHourMin) {
    return controlName === 'hour' ? 'cron-picker.errors.hour-error' : 'cron-picker.errors.minute-error';
  }
  ```